### PR TITLE
fix reload network endpoints

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/42818

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ramaraochavali** [comment]
This regressed as part of this https://github.com/istio/istio/issues/42818. Sorry about that

File: pilot/pkg/model/network.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (mgr *NetworkManager) reloadNetworkEndpoints() {
 		for _, networkconf := range oldNetworks.Networks {
 			for _, ne := range networkconf.Endpoints {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/network.go:125]: better to rename `ne` as well

Comment (ramaraochavali) [reply pilot/pkg/model/network.go]: Done.

**Review by hzxuzhonghu — APPROVED** [review]
LGTM from correctness part

**Review by howardjohn — APPROVED** [review]

**Pull request merged**